src/ApiService/ApiService/onefuzzlib/ScalesetOperations.cs

```diff
@@ -693,33 +695,32 @@ where x.State.ReadyForReset()
             return;
         }
 
-        HashSet<Guid> machineIds = new();
+        var nodesToDelete = new List<Node>();
         foreach (var node in nodes) {
             if (node.DebugKeepNode) {
-                _log.Warning($"not deleting manually overriden node {node.MachineId:Tag:MachineId} in scaleset {scaleset.ScalesetId:Tag:ScalesetId}");
+                _log.Warning($"not deleting manually overridden node {node.MachineId:Tag:MachineId} in scaleset {scaleset.ScalesetId:Tag:ScalesetId}");
             } else {
-                _ = machineIds.Add(node.MachineId);
+                nodesToDelete.Add(node);
             }
         }
 
         switch (disposalStrategy) {
             case NodeDisposalStrategy.Decommission:
                 _log.Info($"decommissioning nodes");
-                await Async.Task.WhenAll(nodes
-                    .Where(node => machineIds.Contains(node.MachineId))
+                await Async.Task.WhenAll(nodesToDelete
                     .Select(async node => {
                         await _context.NodeOperations.ReleaseScaleInProtection(node).IgnoreResult();
                     }));
                 return;
 
             case NodeDisposalStrategy.ScaleIn:
-                _log.Info($"deleting nodes {scaleset.ScalesetId:Tag:ScalesetId} {string.Join(", ", machineIds):Tag:MachineIds}");
-                await _context.VmssOperations.DeleteNodes(scaleset.ScalesetId, machineIds);
-                await Async.Task.WhenAll(nodes
-                    .Where(node => machineIds.Contains(node.MachineId))
+                _log.Info($"deleting nodes {scaleset.ScalesetId:Tag:ScalesetId} {string.Join(", ", nodesToDelete.Select(n => n.MachineId)):Tag:MachineIds}");
+                await _context.VmssOperations.DeleteNodes(scaleset.ScalesetId, nodesToDelete);
+                await Async.Task.WhenAll(nodesToDelete
                     .Select(async node => {
                         await _context.NodeOperations.Delete(node);
```

Comment:
we now have a [DeleteAll](https://github.com/chkeita/onefuzz/blob/49ca5b524802cb370394b654956504d91fc3ebc7/src/ApiService/ApiService/onefuzzlib/orm/Orm.cs#L159) operation in the ORM

Reply:
I'll update to this separately, I think